pkg/utils/patch.go

```diff
@@ -42,12 +42,12 @@ func PatchVolumeSnapshot(
 		return updatedSnapshot, err
 	}
 
-	newSnapshot, err := client.SnapshotV1().VolumeSnapshots(updatedSnapshot.Namespace).Patch(context.TODO(), existingSnapshot.Name, types.MergePatchType, patch, metav1.PatchOptions{}, subresources...)
+	_, err = client.SnapshotV1().VolumeSnapshots(updatedSnapshot.Namespace).Patch(context.TODO(), existingSnapshot.Name, types.MergePatchType, patch, metav1.PatchOptions{}, subresources...)
 	if err != nil {
 		return updatedSnapshot, err
 	}
 
-	return newSnapshot, nil
+	return updatedSnapshot, nil
```

Comment:
I thought "newSnapshot" is the latest one returned from API server?  Why returning "updatedSnapshot"?

Reply:
this was a test commit - I was trying to see if this reduced the 'modified object' error count.

Reply:
Reverting the above change as it did not lower the error count.

Reply:
Can you take a look of the logs and see if you can tell why we got the "modified object" error?

Reply:
Asked SIG API machinery per your suggestion to try and find out. I'll also check the logs to see under what scenarios we face this. The error counts are fairly consistent.